### DISCUSSION TITLE
Dependency Injection

### DISCUSSION THREAD

**Discussion by JustSteveKing** [minicli/minicli#77] [Ideas]
Currently we are using a very basic form of dependency injection in minicli, using magic methods to try and call resolved object. I think we can improve upon this, but there are a few different ways.

1. We build our own DI container, minimal of course, that we can add to replace the simpler approach.
2. We pull in a more minimal package such as PHP-DI to handle this for us.

I would definitely like your opinion on this @erikaheidi as it is right on the edge for if we should allow it or not. But, for resolving services we would get much more benefit from using an actual container.

**Comment by WendellAdriel** [comment]
I think that it depends if the idea is to keep dependency-free, then I think it would be better to create our own minimal container

**Reply by JustSteveKing** [reply]
https://github.com/minicli/minicli/pull/78

**Reply by WendellAdriel** [reply]
I'll check this now

**Comment by erikaheidi** [comment]
Yes, we should keep minicli/minicli dependency-free - just +1'ing what @WendellAdriel said :)

**Comment by WendellAdriel** [comment]
This was already merged on #78
